axelor-base/src/main/java/com/axelor/apps/base/service/LanguageServiceImpl.java

```diff
@@ -0,0 +1,177 @@
+      Language language, String oldName, String oldCode, boolean isNew) {
+    MetaSelect metaSelect;
+    MetaSelectItem metaSelectItem;
+    if (!isNew) {
+      metaSelectItem = findMetaSelectItem(oldName, oldCode);
+      if (metaSelectItem == null) return;
+      metaSelectItem.setTitle(language.getName());
+      metaSelectItem.setValue(language.getCode());
+      metaSelectItemRepository.save(metaSelectItem);
+    } else {
+      // new metaSelectItem
+      Query<MetaSelect> metaSelectQuery =
+          metaSelectRepository
+              .all()
+              .filter("self.name = :selectName and self.module = :newModuleName");
+      metaSelectQuery
+          .bind("selectName", SELECT_NAME)
+          .bind("newModuleName", NEW_SELECT_LANGUAGE_MODULE);
+      metaSelect = metaSelectQuery.fetchOne();
+      if (metaSelect == null) {
+        metaSelect = copyMetaSelect();
+      }
+      metaSelectItem = new MetaSelectItem();
+      metaSelectItem.setOrder((int) getCountMetaSelectItemFromQuery());
+      metaSelectItem.setTitle(language.getName());
+      metaSelectItem.setValue(language.getCode());
+      metaSelect.addItem(metaSelectItem);
+      metaSelectRepository.save(metaSelect);
+    }
+  }
+
+  /**
+   * Find the metaSelectItem according to Langauge obj's name and code.
+   *
+   * @param name
+   * @param code
+   * @return metaSelectItem
+   */
+  public MetaSelectItem findMetaSelectItem(String name, String code) {
+    Query<MetaSelectItem> metaSelectItemQuery =
+        metaSelectItemRepository
+            .all()
+            .filter(
+                "self.title = :oldName and self.value = :oldCode and self.select.module = :moduleName");
+    metaSelectItemQuery
+        .bind("oldName", name)
+        .bind("oldCode", code)
+        .bind("moduleName", NEW_SELECT_LANGUAGE_MODULE);
+    return metaSelectItemQuery.fetchOne();
+  }
+
+  /**
+   * Count the number of metaSelectItem
+   *
+   * @return long : count number
+   */
+  public long getCountMetaSelectItemFromQuery() {
+    Query<MetaSelectItem> metaSelectItemQuery =
+        metaSelectItemRepository.all().filter("self.select.name = :selectName");
+    metaSelectItemQuery.bind("selectName", SELECT_NAME);
+    return metaSelectItemQuery.count();
+  }
+
+  /**
+   * When we remove a Language obj, the related record in select.language should also be removed.
+   *
+   * @param language
+   */
+  @Transactional
+  @Override
+  public void removeLanguageLinkedMetaSelectItem(Language language) {
+    MetaSelectItem metaSelectItem = findMetaSelectItem(language.getName(), language.getCode());
+    if (metaSelectItem == null) return;
+    updateOrderNumber(metaSelectItem.getOrder());
+    metaSelectItemRepository.remove(metaSelectItem);
+  }
+
+  /**
+   * Before remove an MetaSelectItem, we have to update other records' order_seq field.
+   * @param orderNumber
+   */
+  @Transactional
+  public void updateOrderNumber(int orderNumber) {
```

Comment:
please change this method visibility to protected, also the `@Transactional` could be removed since this method is already called in a transaction